app/lib/backend_api_logic/metric_extension.rb

```diff
@@ -37,14 +36,20 @@ def hits_extended_system_name_regex
       end
     end
 
+    def extended_system_name
+      parts = [system_name_without_suffix]
+      parts << owner_id if backend_api_metric?
+      parts.compact.join SYSTEM_NAME_SUFFIX_SEPARATOR
+    end
```

Comment:
1. it saves in "parts" the `system_name_without_suffix` which is always a string of the first part of the system_name, even if system_name is empty, then it is `''`
2. It adds to "parts" the `owner_id` if `owner_type == 'BackendApi'`. In this case owner_id cannot be `nil` unless the BackendApi is built but not persisted.
3. it returns "parts" as a string separated by "-", having removed first the `nil` values of part. But not the empty strings, so it means that if the owner_id is `nil`, it returns only the first part of "parts" (the result of `system_name_without_suffix`), but if it is the other way around, that `system_name_without_suffix` is an empty string, and owner_id has a value, it will return something like '-3' :joy: I just hope that this cannot happen.

Reply:
This is fine.